libs/langchain-aws/src/tests/chat_models.int.test.ts

```diff
@@ -469,3 +469,56 @@ test("Test reasoning_content blocks multiturn streaming", async () => {
   // test a second time to make sure that we've got input translation working correctly
   await doStreaming(streamingMessages);
 });
+
+test("Test ChatBedrockConverse can respond to tool invocations with thinking enabled", async () => {
+  const model = new ChatBedrockConverse({
+    ...baseConstructorArgs,
+    model: "us.anthropic.claude-3-7-sonnet-20250219-v1:0",
+    maxTokens: 5000,
+    additionalModelRequestFields: {
+      thinking: { type: "enabled", budget_tokens: 2000 },
+    },
+  });
+  const tools = [
+    tool(
+      ({ location }: { location: string }) =>
+        `In ${location}, the clouds are heavy with the promise of rain.`,
+      {
+        name: "weather_poet",
+        description:
+          "Gets the current weather conditions for the location, written in a poetic manner.",
+        schema: z.object({
+          location: z.string().describe("Location to get the weather for"),
+        }),
+      }
+    ),
+  ];
+  const modelWithTools = model.bindTools(tools);
+  const messages = [new HumanMessage("What is the current weather in London?")];
+
+  const result = await modelWithTools.invoke(messages);
+  messages.push(result);
+
+  expect(result.tool_calls).toBeDefined();
+  expect(result.tool_calls).toHaveLength(1);
+  // console.log("result.tool_calls?.[0]", result.tool_calls?.[0]);
+
+  expect(typeof result.tool_calls![0]).toBe("object");
+  expect(result.tool_calls![0].name).toBe("weather_poet");
+
+  expect(typeof result.tool_calls![0].id).toBe("string");
+  expect(result.tool_calls![0].id!.length).toBeGreaterThan(0);
+
+  expect(typeof result.tool_calls![0].args).toBe("object");
+  expect(result.tool_calls![0].args.location).toBe("London");
```

Comment:
eh, checking this feels like it could get flaky if the model decides to go above and beyond and e.g. specify "London, England". I'll relax this check and just look for `string`.